R-package/R/xgb.DMatrix.R

```diff
@@ -236,24 +287,37 @@ getinfo <- function(object, ...) UseMethod("getinfo")
 #' @rdname getinfo
 #' @export
 getinfo.xgb.DMatrix <- function(object, name, ...) {
+  allowed_int_fields <- 'group'
+  allowed_float_fields <- c(
+    'label', 'weight', 'base_margin',
+    'label_lower_bound', 'label_upper_bound'
+  )
+  allowed_str_fields <- c("feature_type", "feature_name")
+  allowed_fields <- c(allowed_float_fields, allowed_int_fields, allowed_str_fields, 'nrow')
+
   if (typeof(name) != "character" ||
         length(name) != 1 ||
-        !name %in% c('label', 'weight', 'base_margin', 'nrow',
-                     'label_lower_bound', 'label_upper_bound', "feature_type", "feature_name")) {
-    stop(
-      "getinfo: name must be one of the following\n",
-      "    'label', 'weight', 'base_margin', 'nrow', 'label_lower_bound', 'label_upper_bound', 'feature_type', 'feature_name'"
-    )
+        !name %in% allowed_fields) {
+    stop("getinfo: name must be one of the following\n",
+         paste(paste0("'", allowed_fields, "'"), collapse = ", "))
   }
-  if (name == "feature_name" || name == "feature_type") {
+  if (name == "nrow") {
```

Comment:
Is this useful in practice? We have `dim.xgb.DMatrix` exported.

Reply:
Then I guess it's not really a must-have.